packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -62,20 +64,28 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
     }
 
     let text = toString(value);
-    let numeric = isStringUnit ? NaN : toNumber(value);
+    let numeric = isStringUnit ? NaN : anyToNumber(value);
     let prefix: string | undefined = undefined;
     let suffix: string | undefined = undefined;
+    let color: string | undefined = undefined;
+    let percent: number | undefined = undefined;
+
     let shouldFormat = true;
 
     if (mappings && mappings.length > 0) {
-      const mappedValue = getMappedValue(mappings, value);
+      const mappingResult = getValueMappingResult(mappings, value);
+
+      if (mappingResult) {
+        if (mappingResult.state !== undefined) {
+          text = mappingResult.state;
+        }
 
-      if (mappedValue) {
-        text = mappedValue.text;
-        const v = isStringUnit ? NaN : toNumber(text);
+        if (mappingResult.value !== undefined && !isStringUnit) {
+          numeric = mappingResult.value;
+        }
```

Comment:
Is this really necessary?  We could remove the number option entirely now -- that can be supported via transformations if it is really required.  We always do `shouldFormat = false;` so the unit does not matter